bindings/java/src/main/com/apple/foundationdb/FDBDatabase.java

```diff
@@ -60,8 +61,12 @@ public <T> T run(Function<? super Transaction, T> retryable, Executor e) {
 					T returnVal = retryable.apply(t);
 					t.commit().join();
 					return returnVal;
-				} catch (RuntimeException err) {
-					t = t.onError(err).join();
+				} catch (Exception err) {
+					if(err instanceof RuntimeException || err instanceof ExecutionException) {
```

Comment:
Is `err instanceof ExecutionException` possible here? Note that `retryable` here can't throw checked exceptions, and `t.commit().join()` also throws only non-checked exceptions, so I don't think this should be possible.

Note that if we called `t.commit().get()`, then the above could throw either an `ExecutionException` or an `InterruptedException`, both of which we'd need to catch here, but that's currently not the case.

Reply:
I'm using the Java binding from Scala, and it seems that checked exceptions aren't enforced across the Java/Scala boundary. So a `retryable` implemented in Scala can throw checked exceptions into Java (which isn't expecting one)...

Reply:
> convert the `t.commit().join()` call to `t.commit().get()` and then have it explicitly catch the `ExecutionException`

I think this approach would be better because in case of something like `InterruptedException` the user may expect it to propagate to the outer callers without being wrapped.

Reply:
Looks like we are breaking public API by going with the first approach, because the `run` method needs to declare more checked exceptions...

Let's just pass all `Exception to `onError` then.